src/BlazorWebView/src/Maui/Android/BlazorWebChromeClient.cs

```diff
@@ -22,5 +28,81 @@ public override bool OnCreateWindow(Android.Webkit.WebView? view, bool isDialog,
 			// We don't actually want to create a new WebView window so we just return false 
 			return false;
 		}
+
+		public override bool OnShowFileChooser(Android.Webkit.WebView? view, IValueCallback? filePathCallback, FileChooserParams? fileChooserParams)
+		{
+			if (filePathCallback is null)
+			{
+				return base.OnShowFileChooser(view, filePathCallback, fileChooserParams);
+			}
+
+			_ = CallFilePickerAsync(filePathCallback, fileChooserParams);
+			return true;
+		}
+
+		private static async Task CallFilePickerAsync(IValueCallback filePathCallback, FileChooserParams? fileChooserParams)
+		{
+			var pickOptions = GetPickOptions(fileChooserParams);
+
+			if (fileChooserParams?.Mode == ChromeFileChooserMode.OpenMultiple)
+			{
+				var files = await FilePicker.PickMultipleAsync(pickOptions);
+				if (files is null)
+				{
+					// Task was cancelled, return null to original callback
+					filePathCallback.OnReceiveValue(null);
+					return;
+				}
+
+				var androidUris = files!
+					.Where(f => f is not null)
+					.Select(f => new File(f.FullPath))
+					.Select(Uri.FromFile)
+					.Where(u => u is not null)
+					.Select(u => u!)
+					.ToArray();
```

Comment:
This might be easier to read as an iterative loop (with the added benefit of less allocations):
```suggestion
List<???>? result = null;
foreach (var file in files!)
{
    if (file is null)
    {
        continue;
    }
    
    var uri = Uri.FromFile(new File(file.FullPath);
    if (uri is null)
    {
        continue;
    }
    result ??= new();
    result.Add(uri);
}

var androidUris = result?.ToArray() ? Array.Empty<???>();
```

Reply:
Updated!